extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -96,7 +108,7 @@ protected IcebergTableLocationKey locationKey(
 
                 parquetInstructions = builder.build();
             }
-            return new IcebergTableParquetLocationKey(fileUri, 0, partitions, parquetInstructions);
+            return new IcebergTableParquetLocationKey(fileUri, fileCount++, partitions, parquetInstructions);
```

Comment:
This works because our locationKey creation methods are called from inside a synchronized method.